messaging/src/main/java/org/axonframework/messaging/timeout/AxonTimeLimitedTask.java

```diff
@@ -148,6 +149,67 @@ public void complete() {
         }
     }
 
+    /**
+     * Even though the task was processed successfully, it might have been interrupted while processing, and the
+     * exception might have been caught and swallowed by a lower component. This happens, for example, by the
+     * {@link org.axonframework.eventhandling.LoggingErrorHandler} , which is the default in event processors.
+     * <p>
+     * This function checks if the task was interrupted, and if so, it throws a {@link AxonTimeoutException} to indicate
+     * that the processing was aborted due to a timeout. If the task was not interrupted, it checks if the thread was
+     * interrupted. If it was, it throws an {@link InterruptedException} to indicate that the processing was aborted due
+     * to an interrupt. This effectively restores the proper error status, so upper components can handle it.
+     */
+    public void ensureNoInterruptionWasSwallowed() throws InterruptedException {
+        if (isInterrupted()) {
+            AxonTaskJanitor.LOGGER.info(
+                    "Task [{}] was completed successfully, but was interrupted by the janitor because it was processing for too long. The exception was swallowed by a lower component. Throwing TimeoutException.",
+                    getTaskName()
+            );
+            //noinspection ResultOfMethodCallIgnored
+            Thread.interrupted(); // Clear the interrupt status
+            throw new AxonTimeoutException(String.format("%s has timed out", getTaskName()));
+        } else if (Thread.interrupted()) {
+            // Something was interrupted while processing the task, so we don't need to interrupt it ourselves anymore.
+            this.interrupted = true;
+            this.interruptedExternally = true;
+            Thread.currentThread().interrupt();
+            throw new InterruptedException(String.format("%s was interrupted", getTaskName()));
+        }
+    }
+
+    /**
+     * If an exception is thrown during the handling of the message and it bubbles up, we check if the thread was
+     * interrupted. If it was, we check if the task was interrupted as well. If both hold true, that must mean the
+     * exception was caused by the interruption of the task, and we throw a {@link AxonTimeoutException} to indicate that
+     * the processing was aborted due to a timeout. If the thread was not interrupted, we simply return the original
+     * exception.
+     * <p>
+     * This might happen is someone catches the {@link InterruptedException} and wraps it using a different exception,
+     * or throws a different exception altogether.
+     * <p>
+     * Creators of this task should use this method in their catch block to ensure that the exception is properly
+     * handled and the interrupt status is restored.
+     *
+     * @param e The exception that was thrown during the handling of the message.
+     */
+    public Exception detectInterruptionInsteadOfException(Exception e) {
```

Comment:
Nit: I think it might be good to mention that this exact method is a pure necessity to be invoked when using the `AxonTimeLimitedTask`. So, a short paragraph/sample in the class-level JavaDoc about this, would be a nice addition.